lib/stf/model/device_list.rb

```diff
@@ -12,13 +12,26 @@ def initialize(devices)
     end
 
     def byFilter(filter)
-      filter ? select {|d| d.checkFilter(filter)} : Array.new
+      filter ? select {|d| d.checkFilter(filter)} : []
     end
 
     def exceptFilter(filter)
       filter ? reject {|d| d.checkFilter(filter)} : this
     end
 
+    def healthy(pattern)
+      pattern ? select { |d| d.isHealthy(pattern) } : this
+    end
+
+    # more pessimistic than healthy()
+    def healthyForConnect(pattern)
+      pattern ? select { |d| d.isHealthyForConnect(pattern) } : this
+    end
+
+    def notHealthy(pattern)
```

Comment:
why not just invert ```healthy```? Why do we need this?

Reply:
because it returns list, not boolean. became `select_not_healthy`